graceful-fs.js

```diff
@@ -19,11 +19,44 @@ else if (/\bgfs4\b/i.test(process.env.NODE_DEBUG || ''))
     console.error(m)
   }
 
-if (/\bgfs4\b/i.test(process.env.NODE_DEBUG || '')) {
-  process.on('exit', function() {
-    debug(queue)
-    require('assert').equal(queue.length, 0)
+// Once time initialization
+if (!global[gracefulQueue]) {
+  // This queue can be shared by multiple loaded instances
+  var queue = []
+  Object.defineProperty(global, gracefulQueue, {
+    get: function() {
```

Comment:
This likely needs to be `configurable: true` to allow for 100% coverage during testing.  The problem is that once nyc uses a version with this one time initialization guard it will prevent the local version of fs.close / fs.closeSync from loading.  I'm exploring options but I think test/max-open.js will need to delete this property so it can be recreated by the local graceful-fs.

Reply:
Instead I needed to create a way by for testing of future versions to 'undo' the fs.close / fs.closeSync polyfill, revert to the unpatched functions before replacing with the new functions.  This revert is because if `node_modules/graceful-fs/graceful-fs.js` patched `fs.close` then we would end up running `retry()` provided via node_modules instead of the one provided in this file.

Simply deleting `global[gracefulQueue]` was a bust because of the way `require` works, it actually ends up calling the replacement `fs.closeSync` which called `retry()` in `node_modules/graceful-fs/graceful-fs.js` - this threw an exception since the queue was no longer an array.